code/modules/integrated_electronics/core/printer.dm

```diff
@@ -192,7 +192,8 @@
 		else if(ispath(build_type, /obj/item/integrated_circuit))
 			var/obj/item/integrated_circuit/IC = SScircuit.cached_components[build_type]
 			cost = IC.materials[MAT_METAL]
-		else if(!ispath(build_type, /obj/item/integrated_electronics))
+
+		if(!build_type in SScircuit.circuit_fabricator_recipe_list)
```

Comment:
if((!(build_type in SScircuit.circuit_fabricator_recipe_list["Assemblies"]))&&(!(build_type in SScircuit.circuit_fabricator_recipe_list["Tools"])))

Reply:
AKA if it's a circuit?

Reply:
That's check for assemblys and tools. You handled check if it circuit several lines up.

Reply:
I don't understand why you want me to make a really overtly long check instead of just doing this.